.github/workflows/e2e-test.yaml

```diff
@@ -0,0 +1,82 @@
+name: e2e-test
+permissions:
+  id-token: write
+  contents: read
+on:
+  push:
+    branches:
+      - main
```

Comment:
Can we add a TODO here? e.g, in the future, the repo admin is able to **manually** trigger a e2e test for a specific PR number(git commit)